### PR TITLE
Fix for Bedrock users connecting through Floodgate & DisableAutoUpdateSkin

### PR DESCRIPTION
**Pull request by slamdunx**
SkinStorage updated to work for Bedrock users connecting through Floodgate when using file storage.
SkinStorage also updated so zero expiration prevents updating when player joins.
Linked to issue #323

### CODE REVIEW

File: src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java [hunk 1]
```diff
@@ -320,11 +324,16 @@ public Object getSkinData(String name, boolean updateOutdated) {
...
 
     public Object getSkinData(String name) {
-        return this.getSkinData(name, true);
+        return this.getSkinData(name, false);
     }
 
     private boolean isOld(long timestamp) {
-        return timestamp + TimeUnit.MINUTES.toMillis(Config.SKIN_EXPIRES_AFTER) <= System.currentTimeMillis();
+    	// Check so that updating can be disabled by setting expiration to zero
+    	if (Config.SKIN_EXPIRES_AFTER == 0 ) {
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java:332]: having SKIN_EXPIRES_AFTER on 0 to disable automatic skin update could be misinterpreted fast.
Especially for us having a wide verity of langue speakers.

I think adding another option to disable "automatic update skins"  would be a better option to have it disabled and have 0 always return true

File: src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java [hunk 2]
```diff
@@ -320,11 +324,16 @@ public Object getSkinData(String name, boolean updateOutdated) {
     }
 
     public Object getSkinData(String name) {
-        return this.getSkinData(name, true);
+        return this.getSkinData(name, false);
```
Comment (xknat) [inline src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java:327]: I think this is a unwanted change, we would always want to run the update check on default and have it only not update in some occasions like display (example: skinsgui)

**Review by xknat — CHANGES_REQUESTED** [review]
The changes are great, (filters)
I think the SKIN_EXPIRES_AFTER == 0 is a good check, it only needs to be always updating and have a toggle to disable it to avoid confusion

File: src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java (hunk 2 above)
Comment (slamdunx) [reply src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java:327]: Agreed. Will revert.

File: src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java (hunk 1 above)
Comment (slamdunx) [reply src/main/java/net/skinsrestorer/shared/storage/SkinStorage.java]: SkinAutoUpdates added to config.yml
SKIN_AUTO_UPDATES added to Config.java
isOld method in SkinStorage.java updated to use SKIN_AUTO_UPDATES

**Review by xknat — APPROVED** [review]
Made new Changes, let me know if there is more you want to add so i can merge

**Review by slamdunx — COMMENTED** [review]
Does the comment which is on line 331 of SkinStorage.java in isOld method need to be changed? I should probably have removed the last five words of the comment. Otherwise looks good.

**Comment by xknat** [comment]
@slamdunx like 966312b ?
Let me know if okey like this now

**Review by slamdunx — COMMENTED** [review]
Looks good to me. Thanks.

**Pull request merged**